Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py

```diff
@@ -664,9 +661,30 @@ def list_users_accounts_command(client: Client, args: dict):
 
 
 def format_fetch_start_time_to_timestamp(fetch_start_time: Optional[str]):
-    first_fetch_dt = parse(fetch_start_time).replace(tzinfo=utc)  # type:ignore
-    # Changing 10-digits timestamp to 13-digits by padding with zeroes, since API supports 13-digits
-    return int(first_fetch_dt.timestamp()) * 1000
+    fetch_start_time_datetime = parse(fetch_start_time)  # type: ignore
+    start_fetch_timestamp = fetch_start_time_datetime.timestamp()  # type: ignore
+    if fetch_start_time_datetime.microsecond == 0:  # type: ignore
+        return int(start_fetch_timestamp) * 1000
+
+    timestamp = str(start_fetch_timestamp).replace('.', '')
+    if len(timestamp) == 11:
+        return int(timestamp) * 100
+    elif len(timestamp) == 12:
+        return int(timestamp) * 10
+    elif len(timestamp) == 13:
+        return int(timestamp)
```

Comment:
I'll try to think of a clever way for it, as it can create a bug in the future

Reply:
i agree its not the best practice, however i have unit-tests that check it works as expected in all kinds of forms of timestamps.

see the `test_format_fetch_start_time_to_timestamp` unit-test.

in my case there could be 3 situations:

1. the miliseconds of the date is 0 (multiply by 1000)
2. the miliseconds of that date is of form: <number>00 (multiply by 100)
3. the miliseconds of that date is of form: <number><number>0 (multiply by 10)
4. the miliseconds of that date is of form: <number><number><number> (no need to multiply)

the api knows to work **only** with 13 digits timestamps (which include also the milliseconds)

if you still got a better idea, i'd be happy to hear :)

Reply:
changed